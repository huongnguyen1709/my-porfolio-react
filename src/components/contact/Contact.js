import React, {useState} from 'react';
import emailjs from 'emailjs-com'

import Footer from '../Footer';

const Contact = () => {
    const [formData, setFormData] = useState({
        name: '',
        email: '',
        message: ''
      })

    const sendEmail = (e) => {
        e.preventDefault()

        emailjs.sendForm('service_ro0djge', 'template_x8hk63c', e.target, 'user_zV0aaGfql5j4Y8l0jcyF1')
        .then(res => {
            alert(`Hi ${formData.name}, your message is sent successfully, I will respond as soon as possible. Thank you !`)
        })
        .catch(err => {
            alert(`Hi ${formData.name}, your message is failed to send, please try again. Thank you !`)
        })
        setFormData({
            name: '',
            email: '',
            message: ''
        })  
    }

    return (
        <div className="contact">
            <div className="contact_container">
                <h2 className="contact_title">Get in Touch</h2>
                <p className='contact_info'>Have a question or want to work together ? Contact me via Linkedin or fill in this form, I will respond as soon as possible!</p>

                <form action="form" className="form" onSubmit={sendEmail}>
                    <label className="label-form" htmlFor="name">Name: </label>
                    <br></br>
                    <input 
                        className="form-input" 
                        type="text" 
                        name="name" 
                        placeholder="Enter your name..." 
                        onChange={e => setFormData({...formData, name: e.target.value})}
                        value={formData.name}
                        required 
                    />
                    <br></br>
                    <label className="label-form" htmlFor="email">Email: </label>
                    <br></br>
                    <input 
                        className="form-input" 
                        type="text" 
                        name="user_email"
                        placeholder="Enter your email..." 
                        onChange={e => setFormData({...formData, email: e.target.value})}
                        value={formData.email}
                        required />
                    <br></br>
                    <label className="label-form" htmlFor="email">Message: </label>
                    <br></br>
                    <textarea 
                        className="form-input message-box" 
                        name="message"
                        placeholder="Leave your message..."
                        onChange={e => setFormData({...formData, message: e.target.value})}
                        value={formData.message}
                        required
                    ></textarea>
                    <br></br> <br></br>
                    <button className="btn-send">Send</button>
                </form>

            </div>

            <Footer />
            
        </div>
    );
};

export default Contact;