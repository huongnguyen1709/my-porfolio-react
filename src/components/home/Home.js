import React from 'react';
import {Link} from "react-router-dom";

import NavBar from './NavBar';

import profile from '../../imgs/profile_picture/huong.jpg';
import javascript from '../../imgs/logos/javascript.png';
import react from '../../imgs/logos/react.png';
import nodejs from '../../imgs/logos/nodejs.png';
import mongoDB from '../../imgs/logos/mongoDB.png';

const Header = () => {
    return (
    <div className="home">
         <NavBar />
       <div className="header">
           <div className="huongnguyen">Huong Nguyen</div>  
           <div className="title"><span className="full-stack">Full-Stack</span> Developer</div>
          <Link to="/projects" className="btn-click">View my work</Link>
       </div>
       <img src={profile} alt="huong" className="huong-picture" />

       <div className="line"></div>

       <div className="logo_container">
            <img src={javascript} alt="javascript-logo" className="logo" />
            <img src={react} alt="react-logo" className="logo" />
            <img src={nodejs} alt="typescript-logo" className="logo" />
            <img src={mongoDB} alt="mongoDB-logo" className="logo" />
       </div>

       
    </div>
    );
};

export default Header;