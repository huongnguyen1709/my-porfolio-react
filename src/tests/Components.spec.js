import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

import Home from '../components/home/Home'
import Contact from '../components/contact/Contact'

describe('Home page', () => {
  it('matchs Snapshot', () => {
    const tree = renderer
    .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render items', () => {
    const {container} = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
     expect(container.querySelector('div').classList.contains('home')).toBe(true);
     expect(container.querySelector('div').classList.contains('about')).toBe(false);
  })
})

describe('About page', () => {
  it('should render items', () => {
    render(<Contact />);
    expect(screen.getByText('Get in Touch')).toBeInTheDocument();
    expect(screen.getByText('Send')).toBeInTheDocument();
  })
})
