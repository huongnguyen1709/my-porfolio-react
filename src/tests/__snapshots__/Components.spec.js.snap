// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matchs Snapshot 1`] = `
<div
  className="home"
>
  <div
    className="header_container"
  >
    <div
      className="header"
    >
      <div
        className="name"
      >
        Huong Nguyen
      </div>
      <div
        className="title"
      >
        <span>
          Full-stack
        </span>
         Developer
      </div>
      <a
        className="btn-click"
        href="/about"
        onClick={[Function]}
      >
        Explore me
      </a>
    </div>
    <img
      alt="huongpicture"
      className="profile-picture"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="footer"
  >
    <a
      href="https://www.linkedin.com/in/huongnguyen1709/"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-linkedin icon"
      />
    </a>
    <h5
      className="copyright"
    >
      <i
        className="far fa-copyright"
      />
      2021 Huong Nguyen
    </h5>
    <a
      href="https://github.com/huongnguyen1709"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-github-square github icon"
      />
    </a>
  </div>
</div>
`;
